### ISSUE TITLE
Good places to start contributing

### ISSUE THREAD

**Issue by malcolmbarrett** [clauswilke/sinab#15]
This might be a good first rust project since I'll get exposure to both the C API approach to linking rust code (which I assume is what you're currently using) and the extendr approach. Plus it's a great package. 

Let me know what you think a good place to start familiarizing myself/contributing would be.

PS Do you see sinab replacing/supplementing gridtext?

**Comment by clauswilke** [comment]
Yes, the goal is for sinab to replace gridtext, or rather become the rendering engine for gridtext. The rendering engine that gridtext uses is basically end-of-life. There's no point in doing further development there, it requires a complete rewrite.

The first step will be to replace this C code with Rust code:
https://github.com/clauswilke/sinab/blob/master/src/render_device.c

 More info later, have to run.

**Comment by clauswilke** [comment]
See #16 for details. The nice thing is you can make significant headway without touching any sinab code at all. Just write a new Rust crate that implements the required features using extendr, and once this works we can talk about integrating into sinab. In fact, it's a sufficiently general problem that we may want to keep it separate, maybe as its own  subproject under extendr.

**Comment by malcolmbarrett** [comment]
@clauswilke FYI think I'm going to start working on this in a week or two (post-vacation!)

**Comment by clauswilke** [comment]
Sounds good, looking forward to it. I talked to @andy-thomason the other day that we would probably want to have some sort of graphics library as part of the extendr project, probably in a separate crate e.g. extendr-graphics, so it's probably a good idea to plan for that from the outset. We also had some thoughts about API, we can discuss once there's any sort of starting point of code that does something.

**Comment by andy-thomason** [comment]
That sounds really interesting and should help to drive the design.
